docs/Feature Documentation/Smart Objects/index.md

```diff
@@ -0,0 +1,181 @@
+
+# Smart Objects
+
+Smart Objects are a recursive custom dynamic data structure within SDL Core which can be used to easily store and manipulate complex data. Developers can use Smart Objects to create containers for most primitive types, as well as arrays and maps.
+
+## Usage
+
+The current implementation of Smart Objects contains definitions that allow it to store the following data types: bool, int, long, double, char, string (both as char* and std::string), array, and map.
```

Comment:
char* is a subset of "array", i think it is redundant but could be helpful to note...

Reply:
`char*` is never treated as an array. Technically it is an array, but in practice it is stored as and freely converted to and from `std::string`. You can actually make a char array, but it is stored in a different manner.